obspy/core/stream.py

```diff
@@ -3223,6 +3223,16 @@ def _rotate_to_zne(
             would rotate sets of "BHZ", "BH1", "BH2" (and "HHZ", "HH1", "HH2",
             etc.) channels at the same station.
         """
+        # be nice to users that specify e.g. ``components='ZNE'``..
+        # compare http://lists.swapbytes.de/archives/obspy-users/
+        # 2018-March/002692.html
+        if isinstance(components, (str, native_str)):
+            msg = ("'components' is expected to be a list of one or more "
+                   "component code combinations (e.g. `components=['Z12']` or "
+                   "`components=['Z12', '123']`)")
```

Comment:
The warning seems superfluous to me. Once we add this capability we'll not remove it again without good reason and the code works perfectly fine so its IMHO valid code which should raise no warning. I would just change the docstring to state that single strings are also allowed.

Reply:
fair enough, I guess